gammapy/spectrum/energy.py

```diff
@@ -0,0 +1,281 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import print_function, division
+from astropy.units import Quantity
+from astropy.io import fits
+from ..utils.array import array_stats_str
+
+import astropy.units as u
+import numpy as np
+
+__all__ = ['Energy',
+           'EnergyBounds']
+
+
+class Energy(u.Quantity):
```

Comment:
@mhvk or @eteq – Could you please have a look at this?

We're sub-classing `Energy(Quantity)` because we use it so much in Gammapy and want to attach convenience methods (like FITS table I/O or creating log spaced arrays or going back and forth between energy bounds and energy center arrays for histograms with an energy axis.

Do you think it's a good (or at least OK) idea?

Does the implementation (`__new__`, `__array_finalize__`, `__quantity_subclass__`) look OK?

Are there important tests that should be added besides the ones here?
https://github.com/gammapy/gammapy/pull/290/files#diff-3daf55bf63abb5e56fd51c93e9b16ba6R8

@kingj90 was using `astropy.coordinates.Angle` as an example, and it seems to work OK, but we'd appreciate a review by someone that knows how Quantity really works a lot!